File: app/src/main/java/com/somename/producthunt/screen/RecyclerHolder.java
package com.somename.producthunt.screen;


import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.DataSource;
import com.bumptech.glide.load.engine.GlideException;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.target.Target;
import com.somename.producthunt.R;
import com.somename.producthunt.сontent.Post;

public class RecyclerHolder extends RecyclerView.ViewHolder{

    private TextView mName;
    private TextView mTagline;
    private TextView mVotes;
    private ImageView mThumnail;
    private View mItemView;
    private ProgressBar mProgressBar;

    public RecyclerHolder(View itemView) {
        super(itemView);
        this.mItemView = itemView;
        mName = itemView.findViewById(R.id.item_name);
        mTagline = itemView.findViewById(R.id.item_tagline);
        mVotes = itemView.findViewById(R.id.item_votes);
        mThumnail = itemView.findViewById(R.id.item_thumbnail);
        mProgressBar = itemView.findViewById(R.id.item_placeholder);
    }

    public void bind(@NonNull Post post) {
        mName.setText(post.getName());
        mTagline.setText(post.getTagline());
        mVotes.setText(post.getVotesCount());
        String url = post.getThumbnail().getImageUrl();

        Glide
                .with(mItemView)
                .load(url)
                .listener(new RequestListener<Drawable>() {
                    @Override
                    public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<Drawable> target, boolean isFirstResource) {
                        return false;
                    }

                    @Override
                    public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target, DataSource dataSource, boolean isFirstResource) {
                        mProgressBar.setVisibility(View.GONE);
                        return false;
                    }
                })
                .into(mThumnail);
    }
}

File: presentation/src/main/java/com/somename/producthunt/screen/adapters/RecyclerHolder.kt
package com.somename.producthunt.screen.adapters


import android.graphics.drawable.Drawable
import android.support.v7.widget.RecyclerView
import android.view.View

import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target
import com.somename.producthunt.сontent.PostViewModel
import kotlinx.android.synthetic.main.recycler_item.view.*

class RecyclerHolder(private val mItemView: View) : RecyclerView.ViewHolder(mItemView) {

    fun bind(postViewModel: PostViewModel) {
        mItemView.itemName.text = postViewModel.name
        mItemView.itemTagline.text = postViewModel.tagline
        mItemView.itemVotes.text = postViewModel.votesCount
        val url = postViewModel.thumbnailViewModel?.imageUrl

        Glide
                .with(mItemView)
                .load(url)
                .listener(object : RequestListener<Drawable> {
                    override fun onLoadFailed(e: GlideException?, model: Any, target: Target<Drawable>, isFirstResource: Boolean): Boolean {
                        return false
                    }

                    override fun onResourceReady(resource: Drawable, model: Any, target: Target<Drawable>, dataSource: DataSource, isFirstResource: Boolean): Boolean {
                        itemView.itemPlaceholder.visibility = View.GONE
                        return false
                    }
                })
                .into(itemView.itemThumbnail)
    }
}
